0000-settlement-engine/0000-settlement-engine.md

```diff
@@ -0,0 +1,408 @@
+
+The peer's connector MUST extract the message data and forward it to its settlement engine for processing, and MUST NOT forward the ILP packet to any other connectors. When the settlement engine responds with a response message, the connector MUST encode the raw message from the settlement engine within an ILP FULFILL or ILP REJECT, depending upon the code of the response. If the connector was unable to process the request, it MUST respond with an ILP REJECT.
+
+##### PREPARE
+
+- `amount`: `0`
+- `expiresAt`: _Determined by connector_
+- `executionCondition`: `e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`
+- `destination`: `peer.settle`
+- `data`: _Request message from sender settlement engine_
+
+##### FULFILL
+
+- `fulfillment`: `0000000000000000000000000000000000000000000000000000000000000000`
+- `data`: _Response message from recipient settlement engine_
+
+##### REJECT
+
+- `code`: _Determined by connector from HTTP status of forwarded request_
+- `triggeredBy`: `peer.settle`
+- `message`: _Determined by connector_
+- `data`: _Response message from recipient settlement engine, or empty if antecedent failure_
+
+### Motivation
+
+Settlement engines supercede the [Ledger Plugin Interface (LPIv2)](../deprecated/0024-ledger-plugin-interface-2/0024-ledger-plugin-interface-2.md), an earlier abstraction for settlement integrations with Interledger. This new model addresses these issues:
+
+1. Multi-account plugins required logic for handling ILP packets, increasing implementation complexity.
+2. Plugins bundled settlement and bilateral communication functionality together, limiting composability.
+3. JavaScript plugins limited interoperability with non-JavaScript connector implementations.
+4. Plugins operated in the same process as the connector, which limited scaling the two services independently.
+
+## Specification
+
+### Accounts and identifiers
+
+Each account MUST be identified by a unique, [URL-safe](https://tools.ietf.org/html/rfc3986#section-2.3) string.
+
+The settlement engine MUST be responsible for correlating an account identifier to the peer's identity on the shared ledger or settlement system, if required. For separation of concerns between clearing and settlement, the accounting system is NOT RECOMMENDED to have knowledge of the peer's identity on the shared settlement system.
+
+### Units and quantities
+
+An amount of a given asset or currency may be represented using different denominations or units of the asset. For example, one U.S. dollar may be represented as \$1 or 100 cents, and one Bitcoin may be represented as 1 BTC or 100000000 satoshis.
+
+A **standard unit** is the typical unit of value for a particular asset, such as \$1 in the case of U.S. dollars, or 1 BTC in the case of Bitcoin.
```

Comment:
```suggestion
A **standard unit** is the typical unit of value for a particular asset, such as \$1 in the case of U.S. dollars, or 1 BTC in the case of Bitcoin. This value can be anything two peers agree upon, but it is recommended to use commonly accept values for the standard unit.
```

Reply:
Definitely true, but since this is already confusing enough as it is, I'd much prefer to not even suggest that peers can do that... 😆

Reply:
Agree. I think your addition below also touches upon this. I'll delete this suggestion.